client/invoke.go

```diff
@@ -39,9 +39,16 @@ func (c *GRPCClient) invokeServiceWithRequest(ctx context.Context, req *pb.Invok
 	return
 }
 
-func verbToHTTPExtension(verb string) *v1.HTTPExtension {
+func queryAndVerbToHTTPExtension(query string, verb string) *v1.HTTPExtension {
```

Comment:
Rather than passing string and splitting it this should be a map[string]string throughout